src/index.ts

```diff
@@ -419,11 +423,13 @@ export default class ImageTool implements BlockTool {
    * @param tuneName - tune that has been clicked
    */
   private tuneToggled(tuneName: keyof ImageToolData): void {
-    // inverse tune state
-    this.setTune(tuneName, !(this._data[tuneName] as boolean));
+    // check the tune state
+    const currentState = this.ui.isTuneActive(tuneName);
+
+    this.setTune(tuneName, !currentState);
 
     // reset caption on toggle
-    if (tuneName === 'caption' && !this._data[tuneName]) {
+    if (tuneName === 'caption' && !this.ui.isTuneActive(tuneName)) {
```

Comment:
why we depend on style instead of actual data state?

Reply:
The reason why I considered style because the way `caption` in features needs to be interpreted as `boolean` while its a string, leading to the inconsistency in maintaining the `tune` state (On successive enable/disable while resetting the caption). This doesn't feel like a a permanent fix.

So I took a step back and thought of introducing a `withCaption` field which is of type `boolean` like other tune values which are very consistent with tune state.

I also tried to turn the field `caption` into an object:
```
{
    state: boolean,
    value: string,
}
```
But introducing a complex structure compared to a straight forward `boolean` seems unnecessary.